lib/_sendEvent.ts

```diff
@@ -10,9 +10,12 @@ export function makeSendEvents(requestFn: RequestFnType) {
     if (this._userHasOptedOut) {
       return;
     }
-    if (!this._hasCredentials) {
+    const hasCustomCredentials =
+      additionalParams?.headers["X-Algolia-Application-Id"] &&
+      additionalParams?.headers["X-Algolia-API-Key"];
```

Comment:
```suggestion
    const hasCustomCredentials =
      (this.appId && this.apiKey) || (
      additionalParams?.headers["X-Algolia-Application-Id"] &&
      additionalParams?.headers["X-Algolia-API-Key"]);
```

something like this would mean we can drop _hasCredentials completely